contracts/utils/MigrationHelper.sol

```diff
@@ -0,0 +1,58 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import {IBaseRegistrar} from "../ethregistrar/IBaseRegistrar.sol";
+import {INameWrapper} from "../wrapper/INameWrapper.sol";
+import {Controllable} from "../wrapper/Controllable.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MigrationHelper is Ownable, Controllable {
+    IBaseRegistrar public immutable registrar;
+    INameWrapper public immutable wrapper;
+    address public migrationTarget;
+
+    event MigrationTargetUpdated(address indexed target);
+
+    constructor(IBaseRegistrar _registrar, INameWrapper _wrapper) {
+        registrar = _registrar;
+        wrapper = _wrapper;
+    }
+
+    function setMigrationTarget(address target) external onlyOwner {
+        migrationTarget = target;
+        emit MigrationTargetUpdated(target);
+    }
+
+    function migrateNames(
+        address owner,
+        uint256[] memory tokenIds,
+        bytes memory data
+    ) external onlyController {
+        for (uint256 i = 0; i < tokenIds.length; i++) {
+            registrar.safeTransferFrom(
+                owner,
```

Comment:
Is the `owner` the owner of this contract (aka DAO) or the actual name owner? If the latter, you may want to change to `nameowner` or something as `onlyOwner` at `setMigrationTarget` and `owner` here have different semantics.

Reply:
Looks like it's the actual name owner, because the owner of the name is what you need to transfer it

Reply:
`owner` in this context is the owner of the name, yes. It's passed in as a function parameter.